catalog/dags/common/elasticsearch.py

```diff
@@ -175,51 +177,105 @@ def refresh_index(es_host: str, index_name: str):
-    def check_if_alias_exists(alias: str, es_host: str):
-        """Check if the alias already exists."""
+    @task
+    def get_existing_index(es_host: str, target_alias: str):
+        """Get the index to which the target alias currently points, if it exists."""
+        if not target_alias:
+            raise AirflowSkipException("No target alias was provided.")
+
         es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
-        return (
-            "point_alias.remove_existing_alias"
-            if es_conn.indices.exists_alias(name=alias)
-            else "point_alias.point_new_alias"
-        )
+
+        try:
+            response = es_conn.indices.get_alias(name=target_alias)
+            if len(response) > 1:
+                raise ValueError(
+                    "Expected at most one existing index with target alias"
+                    f"{target_alias}, but {len(response)} were found."
+                )
+            return list(response.keys())[0]
+        except NotFoundError:
+            logger.info(f"Target alias {target_alias} does not exist.")
+            return None
 
     @task
-    def remove_existing_alias(alias: str, es_host: str):
-        """Remove the given alias from any indices to which it  points."""
+    def remove_alias_from_existing_index(
+        es_host: str,
+        index_name: str,
+        target_alias: str,
+    ):
+        """Remove the given alias from the given index."""
+        if not index_name:
+            raise AirflowSkipException(
+                "No existing index from which to remove the target alias."
+            )
+
         es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
         response = es_conn.indices.delete_alias(
-            name=alias,
-            # Remove the alias from _all_ indices to which it currently
-            # applies
-            index="_all",
+            name=target_alias,
+            index=index_name,
         )
         return response.get("acknowledged")
 
     @task
     def point_new_alias(
         es_host: str,
-        index_name: str,
-        alias: str,
+        target_index: str,
+        target_alias: str,
     ):
+        if not target_alias:
+            raise AirflowSkipException("No target alias was provided.")
+
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+        response = es_conn.indices.put_alias(index=target_index, name=target_alias)
+        return response.get("acknowledged")
+
+    @task
+    def delete_old_index(es_host: str, index_name: str, should_delete_old_index: bool):
+        if not should_delete_old_index:
+            raise AirflowSkipException("`should_delete_old_index` is set to `False`.")
+
         es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
-        response = es_conn.indices.put_alias(index=index_name, name=alias)
+        response = es_conn.indices.delete(
+            index=index_name,
+        )
         return response.get("acknowledged")
```

Comment:
I marked the `should_delete_old_index` switch for the DAG with a new alias (a stubborn user test 😸) and it kept retrying due to a clear error: `ValueError: Empty value passed for parameter 'index'`.

 Here, it should catch when the `existing_index` is empty and skip the tasks too (or perhaps make the DAG fail?).

Reply:
Good call! Fixed